shared/src/iosMain/kotlin/org/hyperskill/app/core/domain/model/SwiftyResult.kt

```diff
@@ -0,0 +1,12 @@
+package org.hyperskill.app.core.domain.model
+
+sealed class SwiftyResult<out S, out F : Throwable> {
+    data class Success<out S, out F : Throwable>(val value: S) : SwiftyResult<S, F>()
+    data class Failure<out S, out F : Throwable>(val failure: F) : SwiftyResult<S, F>()
+
+    internal fun toKotlinResult(): Result<S> =
+        when (this) {
+            is Success -> Result.success(value)
+            is Failure -> Result.failure(failure)
+        }
```

Comment:
It's better to extract this kind of functions from data classes. I suggest making it an extension function.